--- /f/tarball/plplot-5.13.0/cmake/modules/FindwxWidgets.cmake
+++ /f/tarball/plplot-5.13.0/cmake/modules/FindwxWidgets.cmake
@@ -213,7 +213,8 @@ if(WIN32 AND NOT CYGWIN AND NOT MSYS AND NOT CMAKE_CROSSCOMPILING)
 else()
   set(wxWidgets_FIND_STYLE "unix")
 endif()
-
+message(STATUS "Findwx: FIND_STYLE= ${wxWidgets_FIND_STYLE}"
+" wxWidgets_FIND_COMPONENTS = wxWidgets_FIND_COMPONENTS}")
 #=====================================================================
 # WIN32_FIND_STYLE
 #=====================================================================
@@ -567,7 +568,9 @@ if(wxWidgets_FIND_STYLE STREQUAL "win32")
       set(WX_LIB_DIR ${wxWidgets_LIB_DIR} CACHE INTERNAL "wxWidgets_LIB_DIR")
       WX_CLEAR_ALL_DBG_LIBS()
       WX_CLEAR_ALL_REL_LIBS()
-    endif()
+        message(STATUS " Here we are, found ? ${WX_ROOT_DIR}/lib/${_WX_TOOL}${_WX_ARCH}_dll ?\n"
+                      " wxWidgets_LIB_DIR= ?${wxWidgets_LIB_DIR}?")
+   endif()
 
     if(WX_LIB_DIR)
       # If building shared libs, define WXUSINGDLL to use dllimport.
@@ -586,7 +589,7 @@ if(wxWidgets_FIND_STYLE STREQUAL "win32")
         endif()
       endforeach()
       DBG_MSG_V("WX_CONFIGURATION_LIST=${WX_CONFIGURATION_LIST}")
-
+        message(STATUS "WX_CONFIGURATION_LIST=${WX_CONFIGURATION_LIST}")
       if(WX_CONFIGURATION)
         set(wxWidgets_FOUND TRUE)
 
@@ -618,7 +621,7 @@ if(wxWidgets_FIND_STYLE STREQUAL "win32")
 
         # Get configuration parameters from the name.
         WX_GET_NAME_COMPONENTS(${wxWidgets_CONFIGURATION} UNV UCD DBG)
-
+        message(STATUS "WX_CONFIGURATION=${WX_CONFIGURATION}")
         # Set wxWidgets lib setup include directory.
         if(EXISTS ${WX_LIB_DIR}/${wxWidgets_CONFIGURATION}/wx/setup.h)
           set(wxWidgets_INCLUDE_DIRS
@@ -644,7 +647,7 @@ if(wxWidgets_FIND_STYLE STREQUAL "win32")
 
         # Settings for requested libs (i.e., include dir, libraries, etc.).
         WX_SET_LIBRARIES(wxWidgets_FIND_COMPONENTS "${DBG}")
-
+        message(STATUS "wxWidgets_FIND_COMPONENTS=${wxWidgets_FIND_COMPONENTS}")
         # Add necessary definitions for unicode builds
         if("${UCD}" STREQUAL "u")
           list(APPEND wxWidgets_DEFINITIONS UNICODE _UNICODE)
@@ -662,6 +665,8 @@ if(wxWidgets_FIND_STYLE STREQUAL "win32")
 #=====================================================================
 else()
   if(wxWidgets_FIND_STYLE STREQUAL "unix")
+   message(STATUS "Findwx: wxWidgets_USE_PREFIX= ${wxWidgets_USE_PREFIX}")
+
     #-----------------------------------------------------------------
     # UNIX: Helper MACROS
     #-----------------------------------------------------------------
