--- /f/tarball/plplot-5.13.0/cmake/modules/FindwxWidgets.cmake
+++ /f/plplot/plplot-5.13.0/cmake/modules/FindwxWidgets.cmake
@@ -766,6 +766,9 @@ else()
 
       # process selection to set wxWidgets_SELECT_OPTIONS
       WX_CONFIG_SELECT_SET_OPTIONS()
+      if(DEFINED wxWidgets_USE_PREFIX)
+        list(APPEND wxWidgets_SELECT_OPTIONS --prefix=${wxWidgets_USE_PREFIX})
+      endif()
       DBG_MSG("wxWidgets_SELECT_OPTIONS=${wxWidgets_SELECT_OPTIONS}")
 
       # run the wx-config program to get cxxflags
